src/libsyntax/parse/parser.rs

```diff
@@ -3232,6 +3247,11 @@ impl<'a> Parser<'a> {
         }
         let not_block = self.token != token::OpenDelim(token::Brace);
         let thn = self.parse_block().map_err(|mut err| {
+            if self.is_identifier_not(&cond) {
+                err.span_suggestion(cond.span, // FIXME: replaced span doesn't incl. whitespace
+                                    "try replacing identifier `not` with the negation operator",
```

Comment:
Looks a bit too wordy for a label, I'd personally use something "try replacing `not` with `!`", but as a minimum the word "identifier" can be removed here.

Reply:
The motivation was to emphasize that `not` has no special meaning in Rust-the-language (it's just another ordinary variable/identifier), even if `rustc`-the-compiler is (after this PR) nice enough to try to figure out notice what you meant, but I agree that _ceteris paribus_, longer messages are worse.